ocaml/xapi/xapi_guest_agent.ml

```diff
@@ -204,9 +204,13 @@ let mutex = Mutex.create ()
    a directory from xenstore. Both are relative to the guest's domainpath. *)
 let get_initial_guest_metrics (lookup: string -> string option) (list: string -> string list) =
   let all_control = list "control" in
-  let to_map kvpairs = List.concat (List.map (fun (xskey, mapkey) -> match lookup xskey with
-      | Some xsval -> [ mapkey, xsval ]
-      | None -> []) kvpairs) in
+  let cant_suspend_reason = lookup "data/cant_suspend_reason" in
```

Comment:
I find this code a bit difficult to parse (because of the `kvpairs` argument to `List.map`) but also have no good suggestion that is much better.

Reply:
Would `kvpairs |> List.map ...` be clearer?